src/gui/widgets/text_box_base.cpp

```diff
@@ -630,21 +660,26 @@ void text_box_base::signal_handler_sdl_key_down(const event::ui_event event,
 
 		case SDLK_RETURN:
 		case SDLK_KP_ENTER:
-			if(!is_composing() || (modifier & (KMOD_CTRL | KMOD_ALT | KMOD_GUI | KMOD_SHIFT))) {
-				return;
-			}
-			// The IME will handle it, we just need to make sure nothing else handles it too.
-			handled = true;
+
+//			if(!is_composing()) {
+//				return;
+//			}
+
+			handle_key_enter(modifier, handled);
 			break;
 
 		case SDLK_ESCAPE:
-			if(!is_composing() || (modifier & (KMOD_CTRL | KMOD_ALT | KMOD_GUI | KMOD_SHIFT))) {
+			if(!is_composing() || (modifier != 0)) {
```

Comment:
Although `!= 0` looks like a simplification, it means this won't show up in a `git grep KMOD_ALT` search. Is there a reason for changing it?

Reply:
Changed to previous version.

Reply:
I still prefer my version, as it catches every possible modifier including ones you didn't think of.